packages/codemodel.go/src/client.ts

```diff
@@ -10,155 +10,145 @@ import * as param from './param.js';
 import * as result from './result.js';
 import * as type from './type.js';
 
-// Client is an SDK client
+/** an SDK client */
 export interface Client {
+  /** the name of the client */
   name: string;
 
+  /** any docs for the client */
   docs: type.Docs;
 
-  // the client options type. for ARM, this will be a QualifiedType (arm.ClientOptions)
+  /** the client options type. for ARM, this will be a QualifiedType (arm.ClientOptions) */
   options: ClientOptions;
 
-  // constructor params that are persisted as fields on the client, can be empty
+  /** constructor params that are persisted as fields on the client, can be empty */
   parameters: Array<param.Parameter>;
 
-  // all the constructors for this client, can be empty
+  /** all the constructors for this client, can be empty */
   constructors: Array<Constructor>;
 
-  // contains client methods. can be empty
-  methods: Array<Method | LROMethod | PageableMethod | LROPageableMethod>;
+  /** contains client methods. can be empty */
+  methods: Array<MethodType>;
```

Comment:
Does switching to just using MethodType break anything? Not familiar with the downstream consequences